api/client/login.go

```diff
@@ -78,11 +78,12 @@ func (cli *DockerCli) promptWithDefault(prompt string, configDefault string) {
 
 func (cli *DockerCli) configureAuth(flUser, flPassword, flEmail, serverAddress string) (types.AuthConfig, error) {
 	authconfig, ok := cli.configFile.AuthConfigs[serverAddress]
+	authconfig.Username = strings.TrimSpace(authconfig.Username)
```

Comment:
I think this line needs to come after the `if !ok {` check, so down to line 84, otherwise authConfig might not be set.